lib/osml/stac_catalog/sc_dataplane.ts

```diff
@@ -0,0 +1,242 @@
+    public STAC_FASTAPI_DESCRIPTION: string = "A STAC FastAPI with an OpenSearch backend",
+    public STAC_FASTAPI_VERSION: string = "2.4.1",
+    public RELOAD: string = "true",
+    public ENVIRONMENT: string = "local",
+    public WEB_CONCURRENCY: string = "10",
+    public ES_PORT: string = "443",
+    public ES_USE_SSL: string = "true",
+    public ES_VERIFY_CERTS: string = "true",
+    public BACKEND: string = "opensearch"
+  ) {}
+}
+
+/**
+ * Interface representing the properties for the SCDataplane construct.
+ */
+export interface SCDataplaneProps {
+  /**
+   * The OSML deployment account.
+   * @type {OSMLAccount}
+   */
+  account: OSMLAccount;
+
+  /**
+   * The OSML VPC (Virtual Private Cloud) configuration for the STAC server.
+   * @type {OSMLVpc}
+   */
+  osmlVpc: OSMLVpc;
+
+  /**
+   * The Docker image code to use for the lambda function
+   */
+  dockerImageCode: DockerImageCode;
+
+  /**
+   * The security group ID to use for the STAC server (optional).
+   * @type {string | undefined}
+   */
+  securityGroupId?: string;
+
+  /**
+   * The IAM (Identity and Access Management) role to be used for Lambda (optional).
+   * @type {IRole | undefined}
+   */
+  lambdaRole?: IRole;
+
+  /**
+   * Custom configuration for the SCDataplane Construct (optional).
+   * @type {SCDataplaneConfig | undefined}
+   */
+  config?: SCDataplaneConfig;
+}
+
+/**
+ * Represents a construct responsible for deploying an ECR container image
+ * for the STAC catalog Lambda.
+ */
+export class SCDataplane extends Construct {
+  public removalPolicy: RemovalPolicy;
+  public config: SCDataplaneConfig;
+  public dockerImageCode: DockerImageCode;
+  public securityGroup?: ISecurityGroup;
+  public ecrDeployment: OSMLECRDeployment;
+  public lambdaFunction: DockerImageFunction;
+  public lambdaRole: IRole;
+  /**
+   * Creates an instance of SCDataplane.
+   * @param {Construct} scope - The scope/stack in which to define this construct.
+   * @param {string} id - The id of this construct within the current scope.
+   * @param {SCDataplaneProps} props - The properties of this construct.
+   * @returns SCDataplane - The SCDataplane instance.
+   */
+  constructor(scope: Construct, id: string, props: SCDataplaneProps) {
+    super(scope, id);
+
+    // Setup class from base properties
+    this.setup(props);
+
+    const opensearchSecurityGroup = new SecurityGroup(
+      this,
+      "StacCatalogOSSecurityGroup",
+      {
+        vpc: props.osmlVpc.vpc
+      }
+    );
+
+    const osDomain = new Domain(this, "StacCatalogOSDomain", {
+      version: EngineVersion.OPENSEARCH_2_11,
+      nodeToNodeEncryption: true,
+      enforceHttps: true,
+      encryptionAtRest: {
+        enabled: true
+      },
+      vpc: props.osmlVpc.vpc,
+      capacity: {
+        dataNodes: this.config.OS_DATA_NODES
+      },
+      vpcSubnets: [props.osmlVpc.selectedSubnets],
+      removalPolicy: RemovalPolicy.DESTROY,
```

Comment:
this should be flagged for the isProd setting.

Reply:
fixed